sdk-development-guidelines.md

```diff
@@ -0,0 +1,200 @@
+# SDK Development Guidelines
+
+### Preamble
+These are common guidelines for Rakuten teams building Android and/or iOS SDKs/libraries.
+
+#### Terminology:
+
+* The term "library", "SDK", "module" and "SDK module" are used interchangeably in this document.
+* MAY, SHOULD, SHOULD NOT, MUST, MUST NOT, RECOMMENDED are defined in [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt).
+* We distinguish between **production code**, i.e. code that is exported/visible to clients, and **supporting code**, i.e. code that is not exported/visible to clients, e.g. tests, build scripts, utilities.
+* We distinguish between **exported dependencies**, i.e. dependencies that consumers of our modules will automatically get in their projects as well, and **not exported dependencies**, i.e. dependencies that are completely hidden from SDK consumers.
+
+## Guidelines
+* [Project Files & Source Control Management](#project-files-&-source-control-management)
+* [Build](#build)
+* [Programming Language](#programming-language)
+* [Coding Style](#coding-style)
+* [Library Design](#library-design)
+* [Testing](#testing)
+* [Documentation](#documentation)
+* [Versioning](#versioning)
+* [Publishing](#publishing)
+* [Continuous Integration & Deployment](#continuous-integration-&-deployment)
+
+### Project files & Source Control Management
+* MUST be under source control management (SCM). RECOMMENDED source control management is git
+* If you intend your source repo to be public ensure you have received the necessary L3/L2 manager and legal department approval
+* MUST use a .gitignore configuration
+* Source code repositories SHOULD follow the naming convention `[platform]-[library name]`, e.g. `android-analytics`, `java-datastore`, `ios-inapmessaging`, `swift-utils`
+    * Rationale: encoding version information (e.g. alpha/core/stable) in the repository breaks links and integration when a module changes
+    * Exceptions: existing modules that already carry additional information in the repository name & renaming would break too many existing integrations.
+* MUST NOT restrict read access to source code repositories unnecessarily
+* For internally hosted code → add the appropriate mailing list with read access
+* For open source projects → public repo on GitHub
+* For Android <img src="images/android.png" alt="drawing" width="15"/>:
```

Comment:
nit: it may be better having the icon before the "For Android" or "For iOS" text

Reply:
rearranged icon